nvme-print-stdout.c

```diff
@@ -799,14 +804,24 @@ static void stdout_phy_rx_eom_descs(struct nvme_phy_rx_eom_log *log)
 		printf("Bottom: %u\n", le16_to_cpu(desc->bottom));
 		printf("Left: %u\n", le16_to_cpu(desc->left));
 		printf("Right: %u\n", le16_to_cpu(desc->right));
-		printf("Number of Rows: %u\n", le16_to_cpu(desc->nrows));
-		printf("Number of Columns: %u\n", le16_to_cpu(desc->ncols));
-		printf("Eye Data Length: %u\n", le16_to_cpu(desc->edlen));
+		printf("Number of Rows: %u\n", nrows);
+		printf("Number of Columns: %u\n", ncols);
+		printf("Eye Data Length: %u\n", desc->edlen);
 
 		if (NVME_EOM_ODP_PEFP(log->odp))
 			stdout_eom_printable_eye(desc);
 
 		/* Eye Data field is vendor specific */
+		if (edlen == 0)
+			continue;
+
+		/* Hex dump Vendor Specific Eye Data */
+		vsdata = malloc(edlen);
+		vsdataoffset = (nrows * ncols) + sizeof(struct nvme_eom_lane_desc);
+		vsdata = (unsigned char *)((unsigned char *)desc + vsdataoffset);
```

Comment:
and here we leak vsdata. 

BTW, `valgrind` is a very good tool find such leaks, give it a try.